app.py

```diff
@@ -192,28 +192,74 @@ def reset_node_render_path(self, node):
         """
         Reset the render path of the specified node.  This
         will force the render path to be updated based on
-        the current script path and configuraton.
+        the current script path and configuration.
         
         Note, this should really never be needed now that the
         path is reset automatically when the user changes something.
         """
         self.__write_node_handler.reset_render_path(node)
 
-    def convert_to_write_nodes(self):
+    def convert_to_write_nodes(self, show_warning=False, create_folders=False):
         """
         Convert all Shotgun write nodes found in the current Script to regular
         Nuke Write nodes.  Additional toolkit information will be stored on 
         additional user knobs named 'tk_*'
+
+        :param show_warning: Optional bool that sets whether a warning box should be displayed to the user;
+         defaults to False.
+        :param create_folders: Optional bool that sets whether the operation will create the required output folders;
+         defaults to False
         """
-        self.__write_node_handler.convert_sg_to_nuke_write_nodes()
 
-    def convert_from_write_nodes(self):
+        # By default we want to convert the write nodes, unless the warning is shown and the user chooses to abort.
+        continue_with_convert = True
+
+        if show_warning:
+            # defer importing the QT module so the app doesn't require QT unless running this method with the warning.
+            from sgtk.platform.qt import QtGui
+            res = QtGui.QMessageBox.question(None,
```

Comment:
As we discussed before, i think it would be great if we checked for promoted knobs in the config and if such are defined, we show a diffeent warning message explaining that info will be lost in the process.

Reply:
Hey Manne, I thought we opted to not show the dialogs at all if promoted knobs were used in the config? Currently, that is what the code does.
https://github.com/shotgunsoftware/tk-nuke-writenode/pull/29/files#diff-3f41e546893dc64b71aaacad12cad815R302